pkcs11/src/defs.rs

```diff
@@ -6,8 +6,63 @@ pub const CRYPTOKI_VERSION: cryptoki_sys::CK_VERSION = cryptoki_sys::CK_VERSION
     major: CRYPTOKI_VERSION_MAJOR,
     minor: cryptoki_sys::CRYPTOKI_VERSION_MINOR,
 };
-pub const LIB_VERSION: CK_VERSION = CK_VERSION { major: 0, minor: 1 };
-pub const LIB_DESCRIPTION: &str = "Nitrokey PKCS#11 library";
+
+pub const LIB_VERSION_MINOR: u8 = {
+    let s = env!("CARGO_PKG_VERSION_MINOR");
+    let mut idx = 0;
+    let mut acc = 0u8;
+    while idx < s.len() {
+        let Some(temp1) = acc.checked_mul(10u8) else {
+            panic!("Minor version does not fit in u8");
+        };
+        let Some(digit) = s.as_bytes()[idx].checked_sub(b'0') else {
+            panic!("Minor version is not a number");
+        };
+        if digit >= 10 {
+            panic!("Minor version is not a number");
+        }
+        let Some(temp2) = temp1.checked_add(digit) else {
+            panic!("Minor version does not fit in u8");
+        };
+        acc = temp2;
+        idx += 1;
+    }
+    acc
+};
+pub const LIB_VERSION_MAJOR: u8 = {
+    let s = env!("CARGO_PKG_VERSION_MAJOR");
+    let mut idx = 0;
+    let mut acc = 0u8;
+    while idx < s.len() {
+        let Some(temp1) = acc.checked_mul(10u8) else {
+            panic!("Major version does not fit in u8");
+        };
+        let Some(digit) = s.as_bytes()[idx].checked_sub(b'0') else {
+            panic!("Major version is not a number");
+        };
+        if digit >= 10 {
+            panic!("Major version is not a number");
+        }
+        let Some(temp2) = temp1.checked_add(digit) else {
+            panic!("Major version does not fit in u8");
+        };
+        acc = temp2;
+        idx += 1;
+    }
+    acc
+};
```

Comment:
Use a helper function for the u8 parsing?

Reply:
Done